src/Elasticsearch/Endpoints/AbstractEndpoint.php

```diff
@@ -60,14 +64,18 @@
     /**
      * @param Transport $transport
      */
-    public function __construct($transport)
+    public function __construct($transport, RequestFactory $requestFactory = null)
     {
         $this->transport = $transport;
+        $this->requestFactory = $requestFactory;
     }
 
     /**
      * @throws \Exception
      * @return array
+     *
+     * @deprecated Performing a request directly on an endpoint is deprecated since 2.2 and will be removed in 3.0,
+     *             you should instead use the createRequest method
      */
     public function performRequest()
```

Comment:
Ok. :)  I probably just don't understand PSR7 and the ramifications enough (I've only briefly skimmed the spec, and quite a while ago).

As I said before, I'm going to be pretty unavailable for the next two-ish weeks.  I'll try to make myself available if/when you have questions, and when I get back I'll dive into the PSR7 spec more deeply and start helping out with the integration.

Thanks for the PR and working on this! :)